src/com/epam/javatrainings/annotation/DTOValidatorImpl.java

```diff
@@ -0,0 +1,155 @@
+package com.epam.javatrainings.annotation;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class DTOValidatorImpl implements DTOValidator{
+
+    private static String message;
+    public static final Pattern VALID_EMAIL_ADDRESS_REGEX =
+            Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", Pattern.CASE_INSENSITIVE);
+
+
+    @Override
+    public <T> void validate(T t) {
+        message = "Validation " + t.getClass().getName() + "instance:";
+        Class clazz = t.getClass();
+        Field[] fields = clazz.getDeclaredFields();
+
+        for (Field field : fields){
+
+            if (field.isAnnotationPresent(Length.class)){
```

Comment:
you can pass only the object to the processor, and in the processor check if some field has specific annotation and validate them.